src/naksu/ui.go

```diff
@@ -342,133 +378,155 @@ func populateBackupCombobox(backupMedia map[string]string, combobox *ui.Combobox
+func mainUIStatusHandler(currentMainUIStatus mainUIStatusType) { //nolint:gocyclo
+	// The cyclomatic complexity calculated by gocyclo is 20
 
-					ui.QueueMain(func() {
-						comboboxLang.Disable()
-						comboboxNic.Disable()
-						comboboxExtNic.Disable()
+	networkstatus.Update()
 
-						buttonStartServer.Disable()
-						buttonMebShare.Enable()
+	// Check general UI status
+	mainUIEnabled := (currentMainUIStatus == mainUIStatusEnabled)
 
-						buttonGetServer.Disable()
-						buttonSwitchServer.Disable()
-						buttonMakeBackup.Disable()
-						buttonDestroyServer.Disable()
-						buttonRemoveServer.Disable()
-					})
+	boxInstalled := false
+	boxRunning := false
+	netAvailable := false
 
-					lastStatus = newStatus
-				}
+	// Make these checks only if main UI is enable to save time when disabling the UI
+	if mainUIEnabled {
+		boxInstalled, boxRunning, netAvailable = getStatusForMainUIStatusHandler()
+	}
+
+	// Create rule for "enabled" for each button
+	buttonRules := []struct {
+		element *ui.Button
+		enable  bool
+	}{
+		{buttonSelfupdateOn, config.IsSelfUpdateDisabled()},
+		{buttonStartServer, mainUIEnabled && boxInstalled && !boxRunning},
+		{buttonMebShare, true},
+		{buttonMakeBackup, mainUIEnabled && boxInstalled && !boxRunning},
+		{buttonDeliverLogs, mainUIEnabled && true},
+		{buttonInstallAbittiServer, mainUIEnabled && !boxRunning && netAvailable},
+		{buttonInstallExamServer, mainUIEnabled && !boxRunning && netAvailable},
+		{buttonDestroyServer, mainUIEnabled && boxInstalled && !boxRunning},
+		{buttonRemoveServer, true},
+	}
+
+	comboboxRules := []struct {
+		element *ui.Combobox
+		enable  bool
+	}{
+		{comboboxLang, mainUIEnabled && !boxRunning},
+		{comboboxNic, mainUIEnabled && !boxRunning},
+		{comboboxExtNic, mainUIEnabled && !boxRunning},
+	}
+
+	ui.QueueMain(func() {
+		for _, buttonRule := range buttonRules {
+			if buttonRule.enable {
+				buttonRule.element.Enable()
+			} else {
+				buttonRule.element.Disable()
 			}
 		}
-	}()
+
+		for _, comboboxRule := range comboboxRules {
+			if comboboxRule.enable {
+				comboboxRule.element.Enable()
+			} else {
+				comboboxRule.element.Disable()
+			}
+		}
+	})
+}
+
+func getStatusForMainUIStatusHandler() (bool, bool, bool) {
+	// Query box installation status
+	boxInstalled, boxInstalledErr := box.Installed()
+	if boxInstalledErr != nil {
+		log.Debug(fmt.Sprintf("Could not query whether VM is installed: %v", boxInstalledErr))
+	}
+
+	boxInstalled = (boxInstalledErr == nil) && boxInstalled
+
+	// Query box running status
+	boxRunning, boxRunningErr := box.Running()
+	if boxRunningErr != nil {
+		log.Debug(fmt.Sprintf("Could not query whether VM is running: %v", boxRunningErr))
+	}
+
+	boxRunning = (boxRunningErr == nil) && boxRunning
+
+	// Query network (internet) connection status
+	netAvailable := network.CheckIfNetworkAvailable()
+
+	return boxInstalled, boxRunning, netAvailable
 }
 
 // checkAbittiUpdate checks
 // 1) if currently installed box is Abitti
```

Comment:
Täsmällisempi olisi: `if currently installed box is Abitti or if no box is installed`

Reply:
:heavy_check_mark: